src/java.naming/share/classes/com/sun/jndi/ldap/sasl/SaslInputStream.java

```diff
@@ -78,7 +78,7 @@ public int read() throws IOException {
         byte[] inBuf = new byte[1];
         int count = read(inBuf, 0, 1);
         if (count > 0) {
-            return inBuf[0];
+            return inBuf[0] & 0xff;
         } else {
             return -1;
```

Comment:
As a separate follow up, even this else block might need some review on whether count can practically be 0 here and if so whether it's OK to return -1 (implying EOF) in such cases.